### PR TITLE
WID-147. Fix Equation components

### PR DESCRIPTION
**Pull request by romina1601**
Fixed the Equations components (corresponding to classes from tvb.datatypes.equations), as they did not work with default values on the 'parameters' port.

### CODE REVIEW

File: xai_components/utils.py [hunk 1]
```diff
@@ -41,7 +42,10 @@ def set_defaults(component, tvb_datatype):
     for attr in tvb_datatype.declarative_attrs:
         attr_value = getattr(tvb_datatype, attr)
         attr_default = attr_value.default
-        if isclass(attr_default) and issubclass(attr_value.field_type, HasTraits):
+        print(type(attr_default))
```
Comment (liadomide) [inline xai_components/utils.py:45]: Maybe have a log here instead of the print ?

**Review by liadomide — APPROVED** [review]

File: xai_components/utils.py [hunk 2]
```diff
@@ -41,7 +42,10 @@ def set_defaults(component, tvb_datatype):
...
+        if issubclass(tvb_datatype, Equation) and callable(attr_default) and attr_default.__name__ == '<lambda>':
+            attr_default = attr_default()
+        elif isclass(attr_default) and issubclass(attr_value.field_type, HasTraits):
             attr_default = attr_default()
```
Comment (liadomide) [inline xai_components/utils.py:49]: both of these branches do the same thing. Was this intended ?

File: xai_components/utils.py (hunk 1 above)
Comment (romina1601) [reply xai_components/utils.py:45]: will remove it, I don't think we actually need it

File: xai_components/utils.py (hunk 2 above)
Comment (romina1601) [reply xai_components/utils.py:49]: the first one is specifically targeted for the `parameters` attribute of `Equation`. the second one is more general and refers to all the attributes that have classes as default values (ex. the noise attribute from an IntegratorStochastic has the `Additive` class provided as default value). but yes, both branches do the same thing, so I guess I could merge them into one longer `if` condition.

**Review by liadomide — APPROVED** [review]

**Pull request merged**